app/src/main/java/org/fossasia/openevent/app/qrscan/ScanQRPresenter.java

```diff
@@ -50,17 +48,31 @@ private void processBarcode(String barcode) {
         scanQRView.showBarcodeData(barcode);
 
         Observable.fromIterable(attendees)
+            .filter(attendee -> attendee.getOrder() != null)
             .filter(attendee -> (attendee.getOrder().getIdentifier() + "-" + attendee.getId()).equals(barcode))
             .subscribeOn(Schedulers.computation())
             .observeOn(AndroidSchedulers.mainThread())
-            .subscribe(attendee -> scanQRView.onScannedAttendee(attendee));
+            .subscribe(attendee -> {
+                if(scanQRView == null)
+                    return;
+
+                scanQRView.onScannedAttendee(attendee);
+            });
+    }
+
+    public void setAttendees(List<Attendee> attendees) {
+        this.attendees = attendees;
     }
 
     @Override
     public void attach() {
```

Comment:
`attach` method should be opposite of `detach`. That is if view is released in `detach` then it should be reassigned in `attach`.
Please make these changes in other presenters too.